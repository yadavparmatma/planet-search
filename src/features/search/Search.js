import React, {Component} from 'react';
import './Search.css'
import {Button, ButtonToolbar, FormControl, FormGroup} from "react-bootstrap";
import SwapiService from "../../services/SwapiService";
import SearchResult from "./SearchResult";
import {Redirect} from "react-router-dom";
import App from "../../App";

class Search extends Component {
    constructor(props) {
        super(props);

        this.state = {
            searchTerm: '',
            planets: []
        };

        this.searchPlanet = this.searchPlanet.bind(this);
    }

    searchPlanet(e) {
        this.setState({searchTerm: e.target.value});

        SwapiService.getPlanets(this.state.searchTerm).then((response) => {
            this.setState({planets: response['results']});
        }).catch((e) => {
            alert(`Error: ${e.message}`);
        })
    }

    render() {
        if (!this.props.isLoggedIn) {
            return (
                <Redirect to={"/"} component={App}/>
            )
        }
        return (
            <div className="App">
                <header className="App-header">
                    <h1 className="App-title">Welcome</h1>
                </header>
                <div className={"search-wrapper"}>
                    <FormGroup>
                        <FormControl type="text" placeholder="Planet" className={"planet-search"}
                                     onChange={this.searchPlanet}/>
                        <ButtonToolbar>
                            <Button bsSize="large" className={"search-button"}
                                    onClick={this.searchPlanet}>Search</Button>
                        </ButtonToolbar>
                    </FormGroup>
                </div>
                <div className={"planet-container"}>
                    <SearchResult planets={this.state.planets}/>
                </div>
            </div>
        );
    }
}

export default Search;
