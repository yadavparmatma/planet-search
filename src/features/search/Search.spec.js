import React from "react";
import renderer from 'react-test-renderer';
import Search from "./Search";


describe('Search', () => {
    it('should match snapshot', () => {
        const wrapper = renderer.create(<Search isLoggedIn={true}/>);

        expect(wrapper.toJSON()).toMatchSnapshot();
    });
});
