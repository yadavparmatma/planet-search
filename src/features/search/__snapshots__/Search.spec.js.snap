// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should match snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <h1
      className="App-title"
    >
      Welcome
    </h1>
  </header>
  <div
    className="search-wrapper"
  >
    <div
      className="form-group"
    >
      <input
        className="planet-search form-control"
        id={undefined}
        onChange={[Function]}
        placeholder="Planet"
        type="text"
      />
      <div
        className="btn-toolbar"
        role="toolbar"
      >
        <button
          className="search-button btn btn-lg btn-default"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Search
        </button>
      </div>
    </div>
  </div>
  <div
    className="planet-container"
  >
    <div
      className="search-result"
    >
      <ul
        className="planet-list"
      />
    </div>
  </div>
</div>
`;
